xbmc/XBDateTime.cpp

```diff
@@ -827,6 +827,8 @@ void CDateTime::GetAsTm(tm& time) const
   time.tm_hour = st.hour;
   time.tm_min = st.minute;
   time.tm_sec = st.second;
+  // Setting to -1 to indicate we don't know the timezone and let the system infer
+  time.tm_isdst = -1;
```

Comment:
Maybe to have set, as:
```cpp
    KODI::TIME::TimeZoneInformation tz;
    switch (KODI::TIME::GetTimeZoneInformation(&tz))
    {
      case KODI::TIME::KODI_TIME_ZONE_ID_DAYLIGHT:
        time.tm_isdst = 1;
        break;
      case KODI::TIME::KODI_TIME_ZONE_ID_STANDARD:
        time.tm_isdst = 0;
        break;
      case KODI::TIME::KODI_TIME_ZONE_ID_UNKNOWN:
        time.tm_isdst = -1;
        break;
    }
  }
```

To you and also Team Kodi, how much becomes `GetAsTm` called? And how safe/good can it be without an `static CDateTimeSpan timezoneBias;` where e.g. in `CDateTime::GetTimezoneBias(...)` function?

With the static there (https://github.com/xbmc/xbmc/blob/0b87a9a2bed758a3705460bb0ddfc7a40ac7c7e5/xbmc/XBDateTime.cpp#L892) I find a bit error prone in case of winter/summer time change, as the static becomes no more updated on running Kodi :thinking:.